wp-cli/media-regenerate.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Helper for dealing with the media regenerate command
+ */
+namespace Automattic\VIP\Commands;
+
+use WP_CLI;
+
+$runner     = WP_CLI::get_runner();
+$assoc_args = $runner->assoc_args;
+
+$media_regenerate_before_invoke = function() use ( &$runner, &$assoc_args ) {
+	// If skip-delete is not set or not true
+	if ( ! isset( $assoc_args['skip-delete'] ) || true !== $assoc_args['skip-delete'] ) {
+		// add skip-delete to the assoc_args array
+		$assoc_args = wp_parse_args(
+			$assoc_args,
+			[ 'skip-delete' => true ]
+		);
+
+		WP_CLI::line( 'Forcing --skip-delete  flag as it\'s required for VIP File Service' );
+		WP_CLI::line( 'Re-running...' );
+
+		// Run the command with the forced skip-delete argument
+		$runner->run_command( $runner->arguments, $assoc_args );
```

Comment:
In testing, I'm seeing an infinite loop when I run `wp media regenerate --skip-delete=false`:

```
[03-Jan-2022 17:35:40 UTC] PHP Fatal error:  Uncaught Error: Xdebug has detected a possible infinite loop, and aborted your script with a stack depth of '256' frames in phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/DocParser.php:49
Stack trace:
#0 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/DocParser.php(49): preg_match('|^([^@][^\\n]+)\\...', 'VIP Backup User...', Array)
#1 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/DocParser.php(62): WP_CLI\DocParser->get_shortdesc()
#2 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CompositeCommand.php(38): WP_CLI\DocParser->get_longdesc()
#3 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php(124): WP_CLI\Dispatcher\CompositeCommand->__construct(Object(WP_CLI\Dispatcher\CompositeCommand), 'role-backup', Object(WP_CLI\DocParser))
#4 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php(57): WP_CLI\Dispatcher\CommandFactory::create_composite_command(Object(WP_CLI\Dispatcher\CompositeCommand), 'role-backup', '\\VIP_Backup_Use...')
#5 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(532): WP_CLI\Dispatcher\CommandFactory::create('role-backup', '\\VIP_Backup_Use...', Object(WP_CLI\Dispatcher\CompositeCommand))
#6 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/RegisterDeferredCommands.php(78): WP_CLI::add_command('vip role-backup', '\\VIP_Backup_Use...', Array)
#7 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Bootstrap\RegisterDeferredCommands->add_deferred_commands()
#8 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(377): WP_CLI::do_hook('before_run_comm...')
#9 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#10 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#11 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#12 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#13 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#14 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#15 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#16 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#17 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#18 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#19 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#20 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#21 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#22 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#23 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#24 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#25 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#26 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#27 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#28 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#29 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#30 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#31 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#32 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#33 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#34 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#35 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#36 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#37 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#38 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#39 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#40 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#41 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#42 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#43 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#44 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#45 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#46 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#47 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#48 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#49 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#50 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#51 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#52 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#53 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#54 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#55 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m...')
#56 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(399): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#57 /chroot/var/www/wp-content/mu-plugins/wp-cli/media-regenerate.php(25): WP_CLI\Runner->run_command(Array, Array)
#58 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/class-wp-cli.php(313): WP_CLI\Runner->Automattic\VIP\Commands\{closure}()
#59 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(470): WP_CLI::do_hook('before_invoke:m in phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/DocParser.php on line 49
```

Reply:
Thanks for this. I fixed this bug in the most recent commit. Can you have a look?